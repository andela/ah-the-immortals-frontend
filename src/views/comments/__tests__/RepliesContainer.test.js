import React from 'react';
import { Router } from 'react-router-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';
import { FormControl, Form, Button } from 'react-bootstrap';
import Replies from '../RepliesContainer';

describe('Tests RepliesContainer', () => {
  const store = configureStore([thunk])({
    comments: { data: {comments: []}, error: {}}
  });
  const props = {
    createChildCommentAction: jest.fn(),
    id: 3,
    slug: 'string'
  };
  const history = createBrowserHistory();
  const wrapper = mount (
    <Router history={history}>
      <Provider store={store}>
        <Replies {...props} />
      </Provider>
    </Router>
  );
  it('tests mounting of container', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('tests input change in form', () => {
    const form = wrapper.find(Form);
    const input = form.find(FormControl);
    input.simulate('change', {
      preventDefault: jest.fn(),
      target: {
        value: 'Test body',
        name: 'body'
      }
    });
    expect(input.length).toEqual(1);
  });
  it('tests submit of a reply', () => {
    const form = wrapper.find(Form);
    form.simulate('submit', {
      preventDefault: jest.fn(),
    });
    expect(form.length).toEqual(1);
  });

});
