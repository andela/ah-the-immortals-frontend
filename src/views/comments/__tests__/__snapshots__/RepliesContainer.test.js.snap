// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests RepliesContainer tests mounting of container 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(Replies)
      createChildCommentAction={[MockFunction]}
      id={3}
      slug="string"
    >
      <Replies
        createChildCommentAction={[Function]}
        data={
          Object {
            "comments": Array [],
          }
        }
        id={3}
        slug="string"
      >
        <CreateReply
          replySubmit={[Function]}
          replyWrite={[Function]}
          value=""
        >
          <Form
            inline={false}
            onSubmit={[Function]}
          >
            <form
              className=""
              onSubmit={[Function]}
            >
              <Bootstrap(Row)>
                <Row
                  bsPrefix="row"
                  noGutters={false}
                >
                  <div
                    className="row"
                  >
                    <Col>
                      <div
                        className="col"
                      >
                        <FormControl
                          className="input-width"
                          name="body"
                          onChange={[Function]}
                          placeholder="Leave a reply"
                          required={true}
                          value=""
                        >
                          <input
                            className="input-width form-control"
                            name="body"
                            onChange={[Function]}
                            placeholder="Leave a reply"
                            required={true}
                            value=""
                          />
                        </FormControl>
                      </div>
                    </Col>
                    <Col
                      lg="2"
                      xs={true}
                    >
                      <div
                        className="col-lg-2 col"
                      >
                        <Button
                          active={false}
                          disabled={false}
                          type="submit"
                          variant="primary"
                        >
                          <button
                            className="btn btn-primary"
                            disabled={false}
                            type="submit"
                          >
                            Send
                          </button>
                        </Button>
                      </div>
                    </Col>
                  </div>
                </Row>
              </Bootstrap(Row)>
            </form>
          </Form>
        </CreateReply>
      </Replies>
    </Connect(Replies)>
  </Provider>
</Router>
`;
